pkg/ansible/watches/watches.go

```diff
@@ -104,65 +101,66 @@ func parseLabelSelector(dls tempLabelSelector) metav1.LabelSelector {
-		ManageStatus                bool                      `yaml:"manageStatus"`
-		WatchDependentResources     bool                      `yaml:"watchDependentResources"`
-		WatchClusterScopedResources bool                      `yaml:"watchClusterScopedResources"`
-		Blacklist                   []schema.GroupVersionKind `yaml:"blacklist"`
-		Finalizer                   *Finalizer                `yaml:"finalizer"`
-		Selector                    tempLabelSelector         `yaml:"selector"`
-	}
-	var tmp alias
+// Use an alias struct to handle complex types
+type alias struct {
+	Group                       string                    `yaml:"group"`
+	Version                     string                    `yaml:"version"`
+	Kind                        string                    `yaml:"kind"`
+	Playbook                    string                    `yaml:"playbook"`
+	Role                        string                    `yaml:"role"`
+	Vars                        map[string]interface{}    `yaml:"vars"`
+	MaxRunnerArtifacts          int                       `yaml:"maxRunnerArtifacts"`
+	ReconcilePeriod             *metav1.Duration          `yaml:"reconcilePeriod,omitempty"`
+	ManageStatus                *bool                     `yaml:"manageStatus,omitempty"`
+	WatchDependentResources     *bool                     `yaml:"watchDependentResources,omitempty"`
+	WatchClusterScopedResources *bool                     `yaml:"watchClusterScopedResources,omitempty"`
+	Blacklist                   []schema.GroupVersionKind `yaml:"blacklist"`
+	Finalizer                   *Finalizer                `yaml:"finalizer"`
+	Selector                    tempLabelSelector         `yaml:"selector"`
+}
+
+// buildWatch will build Watch based on the values parsed from alias
+func (w *Watch) setValuesFromAlias(tmp alias) error {
 
 	// by default, the operator will manage status and watch dependent resources
-	tmp.ManageStatus = manageStatusDefault
+	if tmp.ManageStatus == nil {
+		tmp.ManageStatus = &manageStatusDefault
+	}
 	// the operator will not manage cluster scoped resources by default.
-	tmp.WatchDependentResources = watchDependentResourcesDefault
-	tmp.MaxRunnerArtifacts = maxRunnerArtifactsDefault
-	tmp.ReconcilePeriod = reconcilePeriodDefault
-	tmp.WatchClusterScopedResources = watchClusterScopedResourcesDefault
-	tmp.Blacklist = blacklistDefault
-	tmp.Selector = tempLabelSelector{}
+	if tmp.WatchDependentResources == nil {
+		tmp.WatchDependentResources = &watchDependentResourcesDefault
+	}
+	if tmp.MaxRunnerArtifacts == 0 {
+		tmp.MaxRunnerArtifacts = maxRunnerArtifactsDefault
+	}
 
-	if err := unmarshal(&tmp); err != nil {
-		return err
+	if tmp.ReconcilePeriod == nil {
+		tmp.ReconcilePeriod = &reconcilePeriodDefault
 	}
 
-	reconcilePeriod, err := time.ParseDuration(tmp.ReconcilePeriod)
-	if err != nil {
-		return fmt.Errorf("failed to parse '%s' to time.Duration: %w", tmp.ReconcilePeriod, err)
```

Comment:
The blackList and the tempLabelSelector has not been covered with tests :-( 
It might be a follow up but we need to cover them. 

c/c @asmacdo I think you implemented it. Could you please help us testing if it still ok after the changes?